simulation/collectors.py

```diff
@@ -225,14 +295,29 @@ def get_agent_collectors(
         household_is_renovating_heating_system,
         household_annual_kwh_heating_demand,
         household_annual_heating_fuel_bill,
-        household_epc_c_upgrade_cost_roof,
-        household_epc_c_upgrade_cost_walls,
-        household_epc_c_upgrade_cost_windows,
-        household_heating_system_total_cost_boiler_gas,
-        household_heating_system_total_cost_boiler_oil,
-        household_heating_system_total_cost_boiler_electric,
-        household_heating_system_total_cost_heat_pump_air_source,
-        household_heating_system_total_cost_heat_pump_ground_source,
+        household_element_upgrade_cost_roof,
+        household_element_upgrade_cost_walls,
+        household_element_upgrade_cost_windows,
+        household_heating_system_costs_unit_and_install_boiler_gas,
+        household_heating_system_costs_unit_and_install_boiler_electric,
+        household_heating_system_costs_unit_and_install_boiler_oil,
+        household_heating_system_costs_unit_and_install_heat_pump_air_source,
+        household_heating_system_costs_unit_and_install_heat_pump_ground_source,
```

Comment:
I remember when we had lots of `agent_collectors` last time it made the simulation quite slow. One way around that this time would be to have a single collector `household_heating_system_costs_unit_and_install` (and similar for the below groups) that extracts the full dictionary.

However, the main "issue" with that idea is then we are working with `pd.DataFrame`'s that have dictionary objects in the values of the dataframe, which is really messy and would require postprocessing.

I'm not sure if there is a better way to handle this, and perhaps we just leave it for now.

Reply:
Yeah, indeed - I'd prefer a flat format over dictionaries-in-dataframes, which is why it's this way for now.